src/sentry/web/frontend/project_settings.py

```diff
@@ -187,6 +200,24 @@ def clean_slug(self):
                                         'using that slug' % other.name)
         return slug
 
+    def clean_token(self):
+        token = self.cleaned_data.get('token')
+        if not token:
+            return
+        token_re = r'^[a-zA-Z0-9-]{1,40}$'
```

Comment:
Alright, I updated the regular expression here to actually accommodate basic auth and allow base64 characters and spaces and a length of 255. Unclear if we'd need more than that, but this seems reasonable now.